code-challenges.test.js

```diff
@@ -12,56 +12,231 @@
+//     })
+// }
+
+// PASS  ./code-challenges.test.js
+// testFunction
+//   ✓ takes an object and determine if the values are divisible by three (1 ms)
+// capArr
+//   ✓ takes an array and makes all words capital (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+// Snapshots:   0 total
+// Time:        0.141 s
+// Ran all test suites.
+
+//Refactor:
+
+const capArr = (array) => { 
+    return array.map((value) => value.charAt(0).toUpperCase() + value.slice(1))
+}
+
+// console.log(capArr(randomNouns1))
+// console.log(capArr(randomNouns2))
+
 // --------------------3) Create a function that takes in a string and logs the index of the first vowel.
 
 // Pseudo code:
 
+//input: a string that logs the index of the first vowel
+//output: a function 
+//iterate through a string and log index of first vowel
+//return index from a string with first vowel using the .includes method to compare with a variable containing vowels.
+
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("firstVowel", () => {
+    it("takes a string and logs index of first vowel", () => {
+        const vowelTester1 = "learn"
+        expect (firstVowel(vowelTester1)).toEqual(1)
+        const vowelTester2 = "academy"
+        expect (firstVowel(vowelTester2)).toEqual(0)
+        const vowelTester3 = "challenges"
+        expect (firstVowel(vowelTester3)).toEqual(2)  
+    })
+
+})
+
 const vowelTester1 = "learn"
 // Expected output: 1
 const vowelTester2 = "academy"
 // Expected output: 0
 const vowelTester3 = "challenges"
 // Expected output: 2
 
+// FAIL  ./code-challenges.test.js
+// testFunction
+//   ✓ takes an object and determine if the values are divisible by three (1 ms)
+// capArr
+//   ✓ takes an array and makes all words capital
+// firstVowel
+//   ✕ takes a string and logs index of first vowel (1 ms)
+
+// ● firstVowel › takes a string and logs index of first vowel
+
+//   ReferenceError: firstVowel is not defined
+
 // b) Create the function that makes the test pass.
+
+// const firstVowel = (string) => {
+//     const vowels = ["a", "e", "i", "o", "u"];
+//     for (let i = 0; i < string.length; i++) {
+//         if (vowels.includes(string[i])){
+//             return i
+//         }
+//     }
+// }
+
+// PASS  ./code-challenges.test.js
+// testFunction
+//   ✓ takes an object and determine if the values are divisible by three (1 ms)
+// capArr
+//   ✓ takes an array and makes all words capital
+// firstVowel
+//   ✓ takes a string and logs index of first vowel
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+// Snapshots:   0 total
+// Time:        0.15 s, estimated 1 s
+// Ran all test suites.
+// ✨  Done in 0.66s.
+
+//Refactor: Struggled to refactor was only able to do after extensive googling. 
+
+// const firstVowel = (string) => {
+//     const vowels = ["a", "e", "i", "o", "u"];
+//     return string.map (let i = 0; i < string.length; i++) {
```

Comment:
Remember that `.map` takes a callback function as an argument! The code you are using is for a traditional `for` loop